here/traffic/README.md

```diff
@@ -1,12 +1,36 @@
 # Traffic Analytics Data
 
+- [Data Pipeline](#data-pipeline)
+    - [Important Tables](#important-tables)
+    - [Maintenance](#maintenance)
 - [here_api.py](#here_apipy)
   - [Setup](#setup)
   - [Using the script](#using-the-script)
     - [download](#download)
     - [upload](#upload)
 - [Loading New Data (Old Method)](#loading-new-data-old-method)
 
+## Data Pipeline 
+
+The HERE data pipeline pulls traffic data everyday at 4PM. There is a daily automated airflow pipeline that uses a BashOpertor to run [`here_api.py`](https://github.com/CityofToronto/bdit_data-sources/blob/master/dags/pull_here.py) on the scheduled time. It sends a request specifying the output and data resolution to the HERE API, unzips the file, pipes the output to `psql`, and runs the `/COPY` command to send the output to a simple view `here.ta_view`. The simple view has an insert trigger that executes the function [`here.here_insert_trigger`](https://github.com/CityofToronto/bdit_data-sources/blob/here_declarative/here/traffic/sql/trigger_here_insert.sql) for each row to redirect and transform input data to the parent `here.ta` table which will then insert into the corresponding partitioned table based on the input date.
+
+### Important Tables
+
+`here.ta`
+
+All Traffic Analytics speed data are stored in partitioned tables under the parent `here.ta`. As we upgraded to PostGreSQL version 10, we were able to transition from inheritance based partitioning to declarative partitioning, see [this PR](https://github.com/CityofToronto/bdit_data-sources/pull/497) for more details on the transitioning process. Declarative partitioning allows for an easier setup and requires less maintenance. `here.ta` use range partitioning and the partition key is `dt`. Data is then partitioned by month by specifying the date range bounds.   
+
+`here.ta_view`
+
+This is a simple view created to redirect data to `here.ta` using triggers. Other than additional column `epoch_min` and the lack of columns `dt` and `tod`, it has all the columns in `here.ta`. An trigger with `INSTEAD OF INSERT` statement is created to redirect data received from psql command `/COPY` to `here.ta` along with transformation of `tx` (timestamp) to `dt` (date) and `tod` (time).
+
+`here.ta_yyyymm`
+
+These are partitioned tables of `here.ta`, named with the suffix of  `_yyyymm` (e.g. Data for August 2020 will be in table `here.ta_202008`). 
+
+## Maintenance
+
+After we transition to declarative partitioning, we would only need to create new monthly tables once every year. The function [`create_yearly_tables(yyyy text)`](https://github.com/CityofToronto/bdit_data-sources/blob/master/here/traffic/sql/function_create_yearly_tables.sql) was created for creating monthly partition tables for the input year. There is an airflow DAG dedicated for all our yearly maintenance which runs this function among others once a year.   
```

Comment:
Can we link the dag `https://github.com/CityofToronto/bdit_congestion/blob/network_agg/dags/generate_congestion_agg.py` used in for yearly maintenance  as well?

Reply:
updated